### DISCUSSION TITLE
k_cycle_get_32() is inaccurate nrf52840

### DISCUSSION THREAD

**Discussion by teijokk** [zephyrproject-rtos/zephyr#33871] [Q&A] [answered]
Hi,

I'm trying to get delta value between irqs using k_cycle_get_32(). Then convert to microseconds using k_cyc_to_us_floor32().
I have measured irq pulses in oscilloscope and its 40us but k_cyc_to_us_floor32() returns 30us.

i'm using internal LF 32khz rc oscillator, but I dont know is this clock source is in use in time measure?
CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC=y 

Is there somekind of clock calibration value or something to related to this problem?

Does anyone have idea what could cause this?

br
Teijo

**Comment by henrikbrixandersen** [comment]
@carlescufi Any ideas?

**Comment by teijokk** [comment, marked as answer]
Hello, I have found solution this issue. Seem that k_cycle_get_32() is inaccurate for this purpose. Proper way to do this is to use nrf52840  GPIOTE IN, PPI, and hw timers. But current problem is that zephyr general gpios and nrfx gpiote libraries cant use together. Maybe i have to change all gpios to use nrfx libraries

**Reply by carlescufi** [reply]
@nordic-krch can probably confirm this.
